main.go

```diff
@@ -124,20 +142,27 @@ func runRequest(r pkg.RequestConfig) (err error) {
 		return errors.Wrap(err, "extracting body")
 	}
 
+	if r.WantsCode != 0 && r.WantsCode != resp.StatusCode {
```

Comment:
Ideally, this assert would be done with the other asserts, I don't think it should have its own separate check.

Reply:
Agreed. I just wanted to get that in to test the HttpStatusFmt thing I added in `internal`